### PR TITLE
Add token adapter mixin, fixes #273

### LINKED ISSUE #273: authorize assertion removed from Ember Simple Auth 2.1.1.
According to the ember-simple-auth repo the authorize assertion should not be used any more:
https://github.com/simplabs/ember-simple-auth/pull/2010

The authorize assertian has been removed from Ember Simple Auth 2.1.1. This breaks the following code:
https://github.com/jpadilla/ember-simple-auth-token/blob/facc3bea47196abc553d0248e3aca8c43254440d/addon/mixins/token-authorizer.js#L39-L48

I'll work on a PR to resolve this if is hasn't been fixed already and I'm just looking in the wrong place.

### PR DESCRIPTION
**Pull request by fenichelar**

### CODE REVIEW

**Comment by fenichelar** [comment]
Fixes #273

**Comment by fenichelar** [comment]
v5.1.0
```
# Changelog

- Add `token-adapter` mixin, fixes #273
```

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "ember-simple-auth-token",
-  "version": "5.0.0",
+  "version": "5.0.1",
```
Comment (jpadilla) [inline package.json]: Should we bump minor instead of patch?

File: tests/unit/mixins/token-adapter-test.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import EmberObject from '@ember/object';
+import TokenAdapterMixin from 'ember-simple-auth-token/mixins/token-adapter';
+import { module, test } from 'qunit';
+
+module('Unit | Mixin | token-adapter', function() {
+  test('it works', function (assert) {
+    const TokenAdapterObject = EmberObject.extend(TokenAdapterMixin);
+    const subject = TokenAdapterObject.create();
+    assert.ok(subject);
+  });
```
Comment (jpadilla) [inline tests/unit/mixins/token-adapter-test.js:10]: could we add a simple test for the `headers` computed property?

**Review by jpadilla — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (fenichelar) [reply package.json]: That makes sense. I will update it to 5.1.0

File: tests/unit/mixins/token-adapter-test.js (hunk 2 above)
Comment (fenichelar) [reply tests/unit/mixins/token-adapter-test.js:10]: Will do

File: package.json (hunk 1 above)
Comment (fenichelar) [reply package.json]: This has been updated.

File: tests/unit/mixins/token-adapter-test.js (hunk 2 above)
Comment (fenichelar) [reply tests/unit/mixins/token-adapter-test.js:10]: I have having some difficulties. Have you every written unit tests for a computed property in a mixin?

Comment (fenichelar) [reply tests/unit/mixins/token-adapter-test.js:10]: I gave up

**Review by jpadilla — APPROVED** [review]

**Pull request merged**